arc_cache.go

```diff
@@ -151,6 +170,8 @@ func (b *arccache) PutMany(bs []blocks.Block) error {
 	}
 	for _, block := range good {
 		b.cacheSize(block.Cid(), len(block.RawData()))
+		// unlock after updating store and cache.
```

Comment:
If `PutMany()` returns error (on line 148), then function returns without unlocking the CIDs. Looks like the error from `PutMany()` should be checked _after_ this unlock look, or run the unlock loop in a deferred function.

Reply:
addressed, thanks.